### ISSUE TITLE
Very hard to follow, Wasted Hours no Better off

### ISSUE THREAD

**Issue by crtjr64** [MicrosoftDocs/msteams-docs#6364]
I'm using Microsoft Teams as CDN which documentation says is fine to use because I don't want to invest in using npm or nodejs. Let's get the POC down first. However, all the instructions is in TypeScript and there's no javaScript. When I tried to access these methods: 

`app.getContext, app.initialize, appInstallDialog.openAppInstallDialog, app.openLink, authentication.authenticate, authentication.getAuthToken, authentication.getUser, authentication.registerAuthenticationHandlers was removed to support using Promises, calendar.openCalendarItem, calendar.composeMeeting, call.startCall, chat.getChatMembers, conversations.openConversation, location.getLocation, location.showLocation, mail.openMailItem, mail.composeMail, pages.backStack.navigateBack, pages.navigateCrossDomain, pages.navigateToTab, pages.tabs.getMruTabInstances, pages.tabs.getTabInstances, pages.getConfig, pages.config.setConfig, pages.backStack.navigateBack, people.selectPeople, teams.fullTrust.getConfigSetting, teams.fullTrust.joinedTeams.getUserJoinedTeams
`

I get the error that the app is not defined. When I used microsoftTeams to access the classes and methods I get the error: Initialization Failed. No Parent window found. I have the website running under personal tab but the code still does not work because I don't see the results and of course I can't debug it from within Teams. I just spent the last 2 hours pouring over your documentation looking for clues on how to use the Microsoft Teams SDK under JavaScript and have not found a clue. So, either I've become a lot less intelligent after 20 years of development or the documentation is fragmented and fruitless.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 726e6240-0526-55be-a668-3f1dc3c471a3
* Version Independent ID: 170d240f-00c7-cddb-d1ef-0dc5f229827e
* Content: [Context interface - Teams](https://docs.microsoft.com/en-us/javascript/api/@microsoft/teams-js/app.context?view=msteams-client-js-latest)
* Content Source: [nodejs/docs-ref-autogen/@microsoft/teams-js/app.Context.yml](https://github.com/OfficeDev/msteams-client-docs-ref-typescript/blob/live/nodejs/docs-ref-autogen/@microsoft/teams-js/app.Context.yml)
* Product: **msteams**
* GitHub Login: @charlesprakash
* Microsoft Alias: **o365devx**

**Comment by Nivedipa-MSFT** [comment]
crtjr64 - In Teams JavaScript client SDK v.2.0 the Callbacks converted to promises.
Teams APIs that previously took a callback parameter have been updated to return a JavaScript [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) object. These include the following APIs:


`app.getContext, app.initialize, appInstallDialog.openAppInstallDialog, app.openLink, authentication.authenticate, authentication.getAuthToken, authentication.getUser, authentication.registerAuthenticationHandlers was removed to support using Promises, calendar.openCalendarItem, calendar.composeMeeting, call.startCall, chat.getChatMembers, conversations.openConversation, location.getLocation, location.showLocation, mail.openMailItem, mail.composeMail, pages.backStack.navigateBack, pages.navigateCrossDomain, pages.navigateToTab, pages.tabs.getMruTabInstances, pages.tabs.getTabInstances, pages.getConfig, pages.config.setConfig, pages.backStack.navigateBack, people.selectPeople, teams.fullTrust.getConfigSetting, teams.fullTrust.joinedTeams.getUserJoinedTeams`


You'll need to update the way your code calls any of these APIs to use Promises. For example, if your code is calling a Teams API like this:


```
import microsoftTeams from "@microsoft/teams-js";

microsoftTeams.getContext((context) => { /* ... */ });
```

Needs to be updated to:


```
import { app, Context } from "@microsoft/teams-js";

app.getContext().then((context) => {
    /*...*/
});
```
...or the equivalent async/await pattern:


```
import { app, Context } from "@microsoft/teams-js";

async function example() {
  const context = await app.getContext();
  /*...*/
}
```

Ref Doc : https://docs.microsoft.com/en-us/microsoftteams/platform/tabs/how-to/using-teams-client-sdk?tabs=javascript%2Cmanifest-teams-toolkit#callbacks-converted-to-promises

**Comment by crtjr64** [comment]
I'm sorry I have not responded but I've got applications to develop. I realize promises were being used and I was using them as promises but it still did not work until I reviewed the Channel and Bot example using react. The error I received after wards occurred because of the react command and I use VueJs and not React and know nothing about React. I will take this example into consideration as it's code could be useful. For some reason the integrity checksum doesn't work for me but when I removed it, it worked. Also, my company's behind the time and still using Internet Explorer due to some packaged applications currently in use so es6 isn't always available. Thus, I can't use imports, exports or async... await

**Comment by crtjr64** [comment]
@ChetanSharma-msft I know that already, my 1st issue is the documentation is not good enough to direct people to how to implement v2.0. My 2nd issue is because the documentation doesn't point to how to use the features and methods in this version, I have to waste time guessing at its usage and I've tried several examples and they haven't worked yet. Pointing me to another example further wastes my time. I need a deep dive into the usage of each method of each class within the Teams SDK. 
Example:
               JavaScriptLib.js
                    Methods: A, B, C, D
                    Description of Methods duties
                    Method A:
                            Uses Parameters (1,2,3)
                             Define Parameters and usage
                             Define Method
                             Define Potential Errors
                             Define Desired outcome

**Comment by ChetanSharma-msft** [comment]
@crtjr64 - Thanks for sharing your feedback.
We will check and share the feedback with documentation team.

**Comment by AE-MS** [comment]
> @ChetanSharma-msft I know that already, my 1st issue is the documentation is not good enough to direct people to how to implement v2.0. My 2nd issue is because the documentation doesn't point to how to use the features and methods in this version, I have to waste time guessing at its usage and I've tried several examples and they haven't worked yet. Pointing me to another example further wastes my time. I need a deep dive into the usage of each method of each class within the Teams SDK. Example: JavaScriptLib.js Methods: A, B, C, D Description of Methods duties Method A: Uses Parameters (1,2,3) Define Parameters and usage Define Method Define Potential Errors Define Desired outcome

Hi @crtjr64! When you say that you are looking for deep dives into the usage of each in the documentation, is this the sort of thing you are looking for?
https://docs.microsoft.com/en-us/javascript/api/@microsoft/teams-js/?view=msteams-client-js-latest
From that SDK reference page, there are subpages for each capability in the teamsjs library, describing what each function does, the input parameters and output parameters, etc.
For example, [this page](https://docs.microsoft.com/en-us/javascript/api/@microsoft/teams-js/app?view=msteams-client-js-latest) has that detail about each function, interface, and structure in the `app` capability. Within that page, there is detail about each method, for example [the `initialize` method](https://docs.microsoft.com/en-us/javascript/api/@microsoft/teams-js/app?view=msteams-client-js-latest#@microsoft-teams-js-app-initialize) including expected values of parameters, the return types, etc.

Is this the sort of documentation you are looking for? As you point out, the documentation is in TypeScript and not JavaScript, but if you ignore the type annotations the function should behave the same way in JS.

**Comment by crtjr64** [comment]
Right, now if you go down to the **Function Details** section, you'll encounter the methods, that can lead to the class in which they belong. Now go down to the first one like: enablePrintCapability(). Its deprecated. So click it's link and it does not go to a definition of the new method, but  reload the same page, in fact, the url **https://docs.microsoft.com/en-us/javascript/api/@microsoft/teams-js/?view=msteams-client-js-latest** is exactly the same as the url already loaded. Under the method is its TypeScript example or definition: **function enablePrintCapability()** So, now after going through this it is a waste of my time. I also, went to the samples and took one of the React examples, stripped it down and tried to use it to create a Teams App and it failed with the error: **"We could not update the Teams Settings, please try again"**. So there's no reliable deep dive nor examples to use where I can get my app running on Teams.

**Comment by AE-MS** [comment]
Thanks for the feedback and I totally understand how that confusing and broken link behavior can be frustrating when you're just trying to understand what the function does! For what it's worth, the documentation link behavior you highlighted is a bug that has been filed on the documentation platform and is currently being actively investigated (FYI @erikadoyle).

As a workaround, you can manually navigate to the non-deprecated version of the method by looking at the capability name listed in the reference. In your `enablePrintCapability` example, the deprecation warning references `teamsCore.enablePrintCapability` and you can navigate directly to that by choosing the "teamsCore" entry from the table of contents pane on the left side. It should take you [here](https://docs.microsoft.com/en-us/javascript/api/@microsoft/teams-js/teamscore?view=msteams-client-js-latest). Once on that page, you'll find the [full detail for the non-deprecated `enablePrintCapability` function](https://docs.microsoft.com/en-us/javascript/api/@microsoft/teams-js/teamscore?view=msteams-client-js-latest#@microsoft-teams-js-teamscore-enableprintcapability).

Does that help you get the function reference information you are looking for?

**Comment by ChetanSharma-msft** [comment]
@crtjr64 - Hope you are doing well !!
Please confirm whether above provided workaround helped you or need any further help here?

**Comment by crtjr64** [comment]
Yes, I was able to use the url sent to me

**Comment by ChetanSharma-msft** [comment]
We are closing this issue for now, Please feel free to reopen it, if required.
